### PR TITLE
Fix (non-slow) tests on GPU (torch)

### PR DESCRIPTION
**Pull request by julien-c**
Fixes ~29 failing tests

Also cf previous commit from december: https://github.com/huggingface/transformers/pull/2055/commits/61978c1dd3f340a545e74537c3dae41a4514e867

(not sure why T5 and the common methods were not failing back then)

### CODE REVIEW

File: src/transformers/modeling_bart.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ def _prepare_bart_decoder_inputs(
             causal_lm_mask = None
         new_shape = (bsz, tgt_len, tgt_len)
         # make it broadcastable so can just be added to the attention coefficients
-        decoder_attn_mask = _combine_masks(decoder_padding_mask, causal_lm_mask, new_shape)
+        decoder_attn_mask = _combine_masks(decoder_padding_mask, causal_lm_mask, new_shape).to(device=input_ids.device)
```
Comment (julien-c) [inline src/transformers/modeling_bart.py]: @sshleifer I have no idea if that's the most correct way to do this, depending on the semantics of the operations around.

File: tests/test_modeling_bart.py [hunk 2]
```diff
@@ -172,7 +172,7 @@ class BartHeadTests(unittest.TestCase):
     vocab_size = 99
 
     def test_lm_forward(self):
-        input_ids = torch.Tensor(
+        input_ids = torch.tensor(
```
Comment (julien-c) [inline tests/test_modeling_bart.py]: @sshleifer For consistency let's use this? @LysandreJik

File: src/transformers/modeling_bart.py (hunk 1 above)
Comment (sshleifer) [reply src/transformers/modeling_bart.py]: Unfortunately, `decoder_attn_mask` can be None occasionally.

**Review by sshleifer — APPROVED** [review]

**Review by LysandreJik — APPROVED** [review]
LGTM, will let @sshleifer look at the BART code.

File: src/transformers/modeling_bart.py (hunk 1 above)
Comment (sshleifer) [reply src/transformers/modeling_bart.py]: not True, actually.

**Pull request merged**